src/move/econia/sources/market.move

```diff
@@ -2964,72 +2958,30 @@ module econia::market {
-        assert!(type_info::type_of<BaseType>() // Assert base type.
-                == order_book_ref_mut.base_type, E_INVALID_BASE);
-        assert!(type_info::type_of<QuoteType>() // Assert quote type.
-                == order_book_ref_mut.quote_type, E_INVALID_QUOTE);
-        // Get market underwriter ID.
-        let underwriter_id = order_book_ref_mut.underwriter_id;
-        // If max base to trade flagged as max possible and a buy,
-        // update to max amount that can be bought. If a sell, update
-        // to all available to sell.
-        if (max_base == MAX_POSSIBLE) max_base = if (direction == BUY)
-            (HI_64 - base_ceiling) else base_available;
-        // If max quote to trade flagged as max possible and a buy,
-        // update to max amount that can spend. If a sell, update
-        // to max amount that can receive when selling.
-        if (max_quote == MAX_POSSIBLE) max_quote = if (direction == BUY)
-            quote_available else (HI_64 - quote_ceiling);
-        range_check_trade( // Range check trade amounts.
-            direction, min_base, max_base, min_quote, max_quote,
-            base_available, base_ceiling, quote_available, quote_ceiling);
-        // Calculate max base and quote to withdraw. If a buy:
-        let (base_withdraw, quote_withdraw) = if (direction == BUY)
-            // Withdraw quote to buy base, else sell base for quote.
-            (0, max_quote) else (max_base, 0);
-        // Withdraw optional base coins and quote coins for match,
-        // verifying base type and quote type for market.
-        let (optional_base_coins, quote_coins) =
-            user::withdraw_assets_internal<BaseType, QuoteType>(
-                user_address, market_id, custodian_id, base_withdraw,
-                quote_withdraw, underwriter_id);
-        // Match against order book, storing optionally modified asset
-        // inputs, base and quote trade amounts, and quote fees paid.
-        let (optional_base_coins, quote_coins, base_traded, quote_traded, fees,
-             _) = match(market_id, order_book_ref_mut, user_address,
-                        custodian_id, integrator, direction, min_base,
-                        max_base, min_quote, max_quote, limit_price,
-                        self_match_behavior, optional_base_coins, quote_coins);
-        // Calculate amount of base deposited back to market account.
-        let base_deposit = if (direction == BUY) base_traded else
-            (base_withdraw - base_traded);
-        // Deposit assets back to user's market account.
-        user::deposit_assets_internal<BaseType, QuoteType>(
-            user_address, market_id, custodian_id, base_deposit,
-            optional_base_coins, quote_coins, underwriter_id);
-        // Return base and quote traded by user, fees paid.
+        let price: u64 = if (direction == ASK) {
+            0
+        } else {
+            MAX_POSSIBLE
```

Comment:
Market orders and swap APIs accept `direction` = `BUY` or `SELL` so I suggest a ternary of the form `let side = if (direction == BUY) BID else ASK`.

Then for the price ternary `let price = if (direction == ASK) MIN_PRICE else HI_PRICE`.

With `MIN_PRICE = 1` as I suggest above, this will require minimum refactors in the limit order function and in the match function.